### DISCUSSION TITLE
define function / my task block

### DISCUSSION THREAD

**Discussion by incredibelles** [pybricks/support#1927] [Q&A]
<img width="224" alt="image" src="https://github.com/user-attachments/assets/16bb884d-fcfa-4c1a-8bcd-59f9ebfe15d3">
What do these designations denote on the my task block?
And when calling a task, can you have the contents call on another my task block (so long as there aren't circular dependencies, of course) ?

**Comment by BertLindeman** [comment]
Hello,

The icon is (my guess) there so you can have an icon to document the "task" of the block.
There is no functionality in it.

After you defined a my task block, the block can be found in the tasks palette like this:
![image](https://github.com/user-attachments/assets/68ad874f-8cc8-4e6f-a794-a0b95049ec70)

If you need arguments (parameters) on your block, press the ">" to add one or more.
The parameter block can be copied (drag).
Picture will make it clear, I named the parameter here "my_text":
![image](https://github.com/user-attachments/assets/f6693b57-b245-4472-a84b-faeb516491dc)

Bert

**Reply by laurensvalk** [reply]
Thanks @BertLindeman for the correct answer.

To add to the remaining question:

> And when calling a task, can you have the contents call on another my task block (so long as there aren't circular dependencies, of course

Yes, you can do that.

**Comment by incredibelles** [comment]
awesome, thank you so much for the quick (and clear) responses!
